### PR TITLE
docs: update FAQ docs

### LINKED ISSUE #6594: docs: Restructure the FAQ section
### Improved or typo fixed.

The current "[FAQ](https://apisix.apache.org/docs/apisix/next/FAQ)" page misses internal hyperlinks and could use some rewriting to enhance readability.

### Solution

Make changes to the doc similar to this PR: https://github.com/apache/apisix-website/pull/893

Please assign this issue to me. I will open a PR to discuss the changes.

### PR DESCRIPTION
**Pull request by navendu-pottekkat**
Signed-off-by: Navendu Pottekkat <navendupottekkat@gmail.com>

### Description

This PR updates the FAQ section to be more readable and consistent. 

Fixes #6594

### CODE REVIEW

**Comment by navendu-pottekkat** [comment]
@juzhiyuan @yzeng25 @avinal Could you review this PR?

File: docs/en/latest/FAQ.md [hunk 1]
```diff
@@ -469,40 +468,47 @@ HTTP/1.1 200 OK
 ......
 ```
 
-The route was created successfully. It means that the modification of `X-API-KEY` takes effect.
+**Note**: By using the default token, you could be exposed to security risks. It is recommended to update it when deploying to a production environment.
```
Comment (spacewander) [inline docs/en/latest/FAQ.md:471]:
```suggestion
**Note**: By using the default token, you could be exposed to security risks. It is required to update it when deploying to a production environment.
```
would that be better?

File: docs/en/latest/FAQ.md [hunk 2]
```diff
@@ -21,74 +21,80 @@ title: FAQ
...
-In addition, Apache APISIX has dynamic routing and hot loading of plug-ins, which is especially suitable for API management under micro-service system.
+- It uses etcd to save and synchronize configurations rather than relational databases like PostgreSQL or MySQL. This eliminates the need for polling, synchronizes the configuration in real-time, makes the code concise and avoids a single point of failure.
+- Supports dynamic routing.
+- Supports [hot loading of Plugins](/docs/apisix/plugins/#hot-reload).
 
-## What's the performance of Apache APISIX?
+## What is the performance impact of using Apache APISIX?
 
-One of the goals of Apache APISIX design and development is the highest performance in the industry. Specific test data can be found here：[benchmark](benchmark.md)
+Apache APISIX delivers the best performance among other API gateways with a single-core QPS of 23,000 with an average delay of 0.6 ms.
```
Comment (yzeng25) [inline docs/en/latest/FAQ.md:40]: Could you double check on the QPS and the latency stats? I remember the stats are 18,000 QPS and 0.2 ms latency on the [GitHub intro page](https://github.com/apache/apisix).

image

File: docs/en/latest/FAQ.md [hunk 3]
```diff
@@ -21,74 +21,80 @@ title: FAQ
...
-Apache APISIX is the highest performance API gateway with a single-core QPS of 23,000, with an average delay of only 0.6 milliseconds.
+Specific results of the performance benchmarks can be found [here](benchmark.md).
 
 ## Does Apache APISIX have a user interface？
 
-Yes. Apache APISIX has an experimental feature called [Apache APISIX Dashboard](https://github.com/apache/apisix-dashboard), which is an independent project apart from Apache APISIX. You can deploy Apache APISIX Dashboard to operate Apache APISIX through the user interface.
+Yes. Apache APISIX has an experimental feature called [Apache APISIX Dashboard](https://github.com/apache/apisix-dashboard), which is independent from Apache APISIX. To work with Apache APISIX through a user interface, you can deploy the Apache APISIX Dashboard.
 
-## Can I write my own plugin?
+## Can I write my on Plugins for Apache APISIX?
```
Comment (yzeng25) [inline docs/en/latest/FAQ.md:48]:
```suggestion
## Can I write my own Plugins for Apache APISIX?
```

File: docs/en/latest/FAQ.md (hunk 2 above)
Comment (yzeng25) [reply docs/en/latest/FAQ.md:40]: I mean either of them or all of them could be right. But we need to keep stats correct.

File: docs/en/latest/FAQ.md [hunk 4]
```diff
@@ -21,74 +21,80 @@ title: FAQ
...
 
-Apache APISIX is based on etcd to save and synchronize configuration, not relational databases such as Postgres or MySQL.
+## How does Apache APISIX differ from other API gateways?
 
-This not only eliminates polling, makes the code more concise, but also makes configuration synchronization more real-time. At the same time, there will be no single point in the system, which is more usable.
+Apache APISIX differs in the following ways:
 
-In addition, Apache APISIX has dynamic routing and hot loading of plug-ins, which is especially suitable for API management under micro-service system.
+- It uses etcd to save and synchronize configurations rather than relational databases like PostgreSQL or MySQL. This eliminates the need for polling, synchronizes the configuration in real-time, makes the code concise and avoids a single point of failure.
+- Supports dynamic routing.
```
[the remark this answers is not in the document]
Comment (navendu-pottekkat) [reply docs/en/latest/FAQ.md:35]: @spacewander I did not change what was already existing there.

What do you suggest we add here instead?

File: docs/en/latest/FAQ.md [hunk 5]
```diff
@@ -21,74 +21,80 @@ title: FAQ
...
 -->
 
-## Why a new API gateway?
+## Why do I need a new API gateway?
 
-There are new requirements for API gateways in the field of microservices: higher flexibility, higher performance requirements, and cloud native.
+As organizations move towards cloud native microservices, there is a need for an API gateway that is performant, flexible, secure and scalable.
 
-## What are the differences between Apache APISIX and other API gateways?
+APISIX outperforms other API gateways in these metrics while delivering functions like dynamic routing, a plugin mechanism, a low code dashboard and more.
```
[the remark this answers is not in the document]
Comment (navendu-pottekkat) [reply docs/en/latest/FAQ.md:28]: @nic-chen What do you suggest we add instead?

File: docs/en/latest/FAQ.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (bisakhmondal) [reply docs/en/latest/FAQ.md:35]: `This eliminates the need for polling` - Let's rephrase this. Polling is one of many options in RDBs but they do support event-based push. For eg., Postgres has 1. logical replication and 2. PG_NOTIFY to support real-time event notification. But they are pretty hard to scale compared to etcd - that might be the issue (at least that's what I think).

File: docs/en/latest/FAQ.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (avinal) [reply docs/en/latest/FAQ.md:28]: Maybe pick theses headers: https://github.com/apache/apisix#features

[the remark this answers is not in the document]
Comment (nic-chen) [reply docs/en/latest/FAQ.md:28]: > Maybe pick theses headers: [apache/apisix#features](https://github.com/apache/apisix#features)

yes, the `Full Dynamic` section

**Comment by navendu-pottekkat** [comment]
@spacewander @nic-chen @yzeng25 @bisakhmondal @avinal I have made the changes as suggestions.

**Review by yzeng25 — APPROVED** [review]
LGTM

**Review by nic-chen — APPROVED** [review]

**Review by juzhiyuan — APPROVED** [review]
LGTM, after this PR got merged, @guoqqqi please process the Chinese version 🙏

**Comment by guoqqqi** [comment]
I will create issues, and find a new contributor to complete it 😄

**Pull request merged**